backend/fms_core/import_tool/handlers/experiment_run.py

```diff
@@ -0,0 +1,67 @@
+from datetime import datetime
+from pandas import pandas as pd
+
+from fms_core.models import ExperimentType, PropertyType, Instrument, Container, Sample
+from ._generic import GenericHandler
+from ...services import create_experiment_run
+
+
+
+
+class ExperimentRunHandler(GenericHandler):
+    def __init__(self, experiment_type_obj, instrument, container, start_date,
+                 sample_rows, properties, protocols_dict, properties_by_name_dict):
+
+        self.experiment_run = {'experiment_type': experiment_type_obj, 'instrument': None, 'container': None,
+                               'process': None, 'start_date': start_date}
+
+        # Get Instrument
+        instrument_name = instrument['name']
+        if instrument_name:
+            try:
+                self.experiment_run['instrument'] = Instrument.objects.get(name=instrument_name)
+            except Exception as e:
+                self.errors["instrument"] = f"No instrument named {instrument_name} could be found."
+
+        # Get or Create Container
+        barcode = container['barcode']
+        kind = container['kind']
+        if barcode and kind:
+            try:
+                self.experiment_run['container'], _ = Container.objects.get_or_create(
```

Comment:
Experiment_run should always create its container ... You don't really want a recycled container...

Reply:
Same thing is done in the resource ExperimentRun...